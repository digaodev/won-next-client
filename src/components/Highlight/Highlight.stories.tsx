import { Story, Meta } from '@storybook/react/types-6-0'

import Highlight, { HighlightProps } from './Highlight'

export default {
  title: 'Highlight',
  component: Highlight,
  argTypes: {},
  args: {
    title: 'Red Dead is back!',
    subtitle: "Come see John's new adventure",
    buttonLabel: 'Buy Now',
    buttonLink: '/buy',
    backgroundImage: '/img/red-dead-img.jpg'
  },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

const Template: Story<HighlightProps> = (args) => (
  <div style={{ maxWidth: '104rem' }}>
    <Highlight {...args} />
  </div>
)

export const Default = Template.bind({})

export const WithFloatImage = Template.bind({})
WithFloatImage.args = {
  floatImage: '/img/red-dead-float.png'
}
