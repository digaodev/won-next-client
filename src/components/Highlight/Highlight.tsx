import Button from 'components/Button/Button'
import * as S from './Highlight.styles'

export type HighlightProps = {
  /**
   * Required title
   */
  title: string

  /**
   * Required subtitle
   */
  subtitle: string

  /**
   * Required button label
   */
  buttonLabel: string

  /**
   * Required button link
   */
  buttonLink: string

  /**
   * Required background image url
   */
  backgroundImage: string

  /**
   * Optional float image url
   */
  floatImage?: string

  /**
   * Optional content alignment
   */
  alignment?: 'right' | 'left'
}

const Highlight = ({
  title,
  subtitle,
  buttonLabel,
  buttonLink,
  backgroundImage,
  floatImage,
  alignment = 'right'
}: HighlightProps) => (
  <S.Wrapper backgroundImage={backgroundImage} alignment={alignment}>
    {!!floatImage && <S.FloatImage src={floatImage} alt={title}></S.FloatImage>}
    <S.Content>
      <S.Title>{title}</S.Title>
      <S.Subtitle>{subtitle}</S.Subtitle>

      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </S.Content>
  </S.Wrapper>
)

export default Highlight
