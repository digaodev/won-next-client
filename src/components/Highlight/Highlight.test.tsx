import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'

import Highlight from './Highlight'
import * as S from './Highlight.styles'

const props = {
  title: 'Heading 1',
  subtitle: 'Heading 2',
  buttonLabel: 'Buy Now',
  buttonLink: '/buy',
  backgroundImage: '/img/red-dead-img.jpg',
  floatImage: '/img/red-dead-float.jpg'
}

describe('<Highlight />', () => {
  it('should render the Highlight', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(
      screen.getByRole('heading', { name: /heading 1/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /heading 2/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /buy/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the Highlight with a background image', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(container.firstChild).toHaveStyle({
      backgroundImage: `url(${props.backgroundImage})`
    })
  })

  it('should render the Highlight with a floating background image', () => {
    renderWithTheme(<Highlight {...props} />)

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      props.floatImage
    )
  })

  it('should render the Highlight with the text right-aligned by default', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(container.firstChild).toHaveStyle({
      'grid-template-areas': "'floatImage content'"
    })

    expect(container.firstChild).toHaveStyleRule('text-align', 'right', {
      modifier: `${S.Content}`
    })
  })

  it('should render the Highlight with the text left-aligned when alignment is passed', () => {
    const { container } = renderWithTheme(
      <Highlight {...props} alignment="left" />
    )

    expect(container.firstChild).toHaveStyle({
      'grid-template-areas': "'content floatImage'"
    })

    expect(container.firstChild).toHaveStyleRule('text-align', 'left', {
      modifier: `${S.Content}`
    })
  })
})
