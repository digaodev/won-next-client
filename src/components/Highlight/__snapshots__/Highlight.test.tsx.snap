// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render the Highlight 1`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  border: 0;
  border-radius: 0.4rem;
  color: #FAFAFA;
  cursor: pointer;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c6:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  position: relative;
  display: grid;
  grid-template-areas: 'floatImage content';
  background-image: url(/img/red-dead-img.jpg);
  background-position: center center;
  background-size: cover;
  height: 23rem;
  grid-template-areas: 'floatImage content';
  grid-template-columns: 1.3fr 2fr;
}

.c0::after {
  position: absolute;
  content: '';
  background-color: rgba(0,0,0,0.6);
  height: 100%;
  width: 100%;
}

.c0 .c2 {
  text-align: right;
}

.c1 {
  grid-area: floatImage;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  max-height: 23rem;
  max-width: 100%;
  z-index: 10;
}

.c3 {
  grid-area: content;
  padding: 1.6rem;
  z-index: 10;
}

.c4 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 600;
}

.c5 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

@media (min-width:768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c1 {
    max-height: 32rem;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2.8rem;
  }
}

<section
  class="c0"
>
  <img
    alt="Heading 1"
    class="c1"
    src="/img/red-dead-float.jpg"
  />
  <div
    class="c2 c3"
  >
    <h2
      class="c4"
    >
      Heading 1
    </h2>
    <h3
      class="c5"
    >
      Heading 2
    </h3>
    <a
      class="c6"
      href="/buy"
    >
      <span>
        Buy Now
      </span>
    </a>
  </div>
</section>
`;
